### PR TITLE
Move PyTorch Models Tests to Azure runners

### PR DESCRIPTION
**Pull request by ababushk**
### Details:
Moves PyTorch Models Tests to Azure GitHub Actions runners to enable models caching on Azure shared storage

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -530,18 +530,14 @@ jobs:
...
   PyTorch_Models_Tests:
     name: PyTorch Models tests
     if: fromJSON(needs.smart_ci.outputs.affected_components).PyTorch_FE.test
     needs: [ Build, Smart_CI, Openvino_tokenizers ]
     uses: ./.github/workflows/job_pytorch_models_tests.yml
     with:
-      runner: ${{ github.event_name == 'schedule' && 'ubuntu-20.04-16-cores' || 'ubuntu-20.04-8-cores' }}
+      runner: ${{ github.event_name == 'schedule' && 'aks-linux-16-cores-32gb' || 'aks-linux-8-cores-16gb' }}
+      container: '{"image": "openvinogithubactions.azurecr.io/dockerhub/ubuntu:20.04", "volumes": ["/mount:/mount"]}'
       event: ${{ github.event_name }}
```
Comment (rkazants) [inline .github/workflows/linux.yml:542]: you also need to do with this: https://github.com/openvinotoolkit/openvino/blob/master/tests/model_hub_tests/models_hub_common/constants.py#L30
set USE_SYSTEM_CACHE and also set HUGGINGFACE_HUB_CACHE.
Besides, I am not sure if need to have common variable for both PyTorch and TensorFlow HuggingFace models. I think we need to split and rename current variable for it.
@mvafin, please comment

Comment (rkazants) [reply .github/workflows/linux.yml:548]: Hi @ababushk, we decided do not split cached TF and PR models now. That is because we have limitation in tf_sentence_transformers package that does not allow to set cache dir via method option for model loading. It requires to use common HF env variable.
So please proceed with your PR.

**Comment by rkazants** [comment]
@mvafin, @popovaan, please review.

File: tests/model_hub_tests/models_hub_common/constants.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
 hf_hub_cache_dir = tempfile.gettempdir()
 if os.environ.get('USE_SYSTEM_CACHE', 'True') == 'False':
     no_clean_cache_dir = True
-    os.environ['HUGGINGFACE_HUB_CACHE'] = hf_hub_cache_dir
+    hf_hub_cache_dir = os.environ['HUGGINGFACE_HUB_CACHE']
```
Comment (mvafin) [inline tests/model_hub_tests/models_hub_common/constants.py:33]: What is the purpose of this change? As I understand `HUGGINGFACE_HUB_CACHE` may not exist in general

Comment (ababushk) [reply tests/model_hub_tests/models_hub_common/constants.py:33]: It isn't very clear for me either.
My understanding was that if `USE_SYSTEM_CACHE` is `False`, we don't need to use tempdir, but we use the path from `HUGGINGFACE_HUB_CACHE` environment variable.
But we already have `HF_HUB_CACHE` and `hf_cache_dir` for that. Am I not aware of some libraries' internals?
